OptimizelySDK/Notifications/NotificationCenter.cs

```diff
@@ -0,0 +1,197 @@
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="featureAccessCallback">Callback function to call when event gets triggered</param>
+        /// <returns></returns>
+        public int AddNotification(NotificationType notificationType, FeatureAccessCallback featureAccessCallback)
+        {
+            return AddNotification(notificationType, (object)featureAccessCallback);
+        }
+
+
+        /// <summary>
+        /// Add a notification callback to the notification center.
+        /// </summary>
+        /// <param name="notificationType">Notification type</param>
+        /// <param name="notificationCallback">Callback function to call when event gets triggered</param>
+        /// <returns></returns>
+        private int AddNotification(NotificationType notificationType, object notificationCallback)
+        {
+            var notificationHoldersList = Notifications[notificationType];
+
+            if (!Notifications.ContainsKey(notificationType) || Notifications[notificationType].Count == 0)
+                Notifications[notificationType][NotificationId] = notificationCallback;
+            else
+            {
+                foreach(var notification in this.Notifications[notificationType])
+                {
+                    if ((Delegate)notification.Value == (Delegate)notificationCallback)
+                        return -1;
+                }
+
+                Notifications[notificationType][NotificationId] = notificationCallback;
+            }
+
+            int retVal = NotificationId;
+
+            NotificationId += 1;
+
+            return retVal;
+        }
+
+        /// <summary>
+        /// Remove a previously added notification callback.
+        /// </summary>
+        /// <param name="notificationId">Id of notification</param>
+        /// <returns>Returns true if found and removed, false otherwise.</returns>
+        public bool RemoveNotification(int notificationId)
+        {
+            foreach(var key in Notifications.Keys)
+            {
+                if (Notifications[key] != null && Notifications[key].Any(notification => notification.Key == notificationId))
+                {
+                    Notifications[key].Remove(notificationId);
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Remove all notifications for the specified notification type.
+        /// </summary>
+        /// <param name="notificationType">The notification type</param>
+        public void ClearNotifications(NotificationType notificationType)
+        {
+            Notifications[notificationType].Clear();
+        }
+
+        /// <summary>
+        /// Removes all notifications.
+        /// </summary>
+        public void ClearAllNotifications()
+        {
+            Notifications.Clear();
```

Comment:
Please run a scenario. 
1 - add notification of any type
2 - clearnotifications
3 - clearnotification of anytype. 
@oakbani  FYI

Reply:
I have updated the ClearAllNotifications method. Please review. Will add that unit test.